commons/src/main/java/fr/free/nrw/commons/upload/ShareActivity.java

```diff
@@ -208,4 +220,35 @@ public boolean onOptionsItemSelected(MenuItem item) {
         return super.onOptionsItemSelected(item);
     }
 
+    /**
+     * Builds URL with image coords for MediaWiki API calls
+     * Example URL: https://commons.wikimedia.org/w/api.php?action=query&prop=categories|coordinates|pageprops&format=json&clshow=!hidden&coprop=type|name|dim|country|region|globe&codistancefrompoint=38.11386944444445|13.356263888888888&
+     * generator=geosearch&redirects=&ggscoord=38.11386944444445|13.356263888888888&ggsradius=100&ggslimit=10&ggsnamespace=6&ggsprop=type|name|dim|country|region|globe&ggsprimary=all&formatversion=2
+     */
+    public static class UrlBuilder {
+        private static String buildUrl (String coords){
```

Comment:
Hm, you might consider just keeping this as a method of MwVolleyApi. I'm not sure putting it in a class helps much since it's stateless.